lib/modules/versioning/rpm/index.ts

```diff
@@ -0,0 +1,290 @@
+    let rpmRelease = '';
+    let rpmPreRelease = '';
+    let snapshot = '';
+    const releaseIndex = remainingVersion.indexOf('-');
+    const prereleaseIndex = remainingVersion.indexOf('~');
+
+    // Note: There can be a snapshot if there is no prerelease. Snapshot always beat no snapshot,
+    // so if there is 3.12.0-1 vs 3.12.0-1^20231110, the snapshot wins.
+    // The logic below only creates snapshot IF there is a prerleease version. This logic is NOT
+    // correct, but the result is still correct due to the caret being ignored in release, and
+    // release continue comparing
+    //
+    // Note: If there IS a tilde preceding the caret, then snapshot DOES NOT win
+    // Example: 3.12.0-1~^20231001 LOSES to 3.12.0-1 and
+    // 3.12.0-1~^20231001 LOSES to 3.12.0-1^20231001
+    const snapshotIndex = remainingVersion.indexOf('^');
+
+    if (releaseIndex >= 0) {
+      upstreamVersion = remainingVersion.slice(0, releaseIndex);
+
+      // Do NOT splice out prerelease, we need to distinguish if the flag is set or not, regardless if there is a version.
+      // The tilde will get filtered out during regex
+      if (prereleaseIndex >= 0) {
+        rpmRelease = remainingVersion.slice(releaseIndex, prereleaseIndex);
+        if (snapshotIndex >= 0) {
+          rpmPreRelease = remainingVersion.slice(
+            prereleaseIndex,
+            snapshotIndex,
+          );
+          snapshot = remainingVersion.slice(snapshotIndex + 1);
+        } else {
+          rpmPreRelease = remainingVersion.slice(prereleaseIndex);
+        }
+      } else {
+        rpmRelease = remainingVersion.slice(releaseIndex + 1);
+      }
+    } else {
+      upstreamVersion = remainingVersion;
+    }
+
+    upstreamVersion;
+
+    const release = [...remainingVersion.matchAll(regEx(/\d+/g))].map((m) =>
+      parseInt(m[0], 10),
+    );
+
+    return {
+      epoch,
+      upstreamVersion,
+      rpmRelease,
+      release,
+      rpmPreRelease,
+      snapshot,
+    };
+  }
+
+  protected _compare_string(s1: string, s2: string): number {
+    if (s1 === s2) {
+      return 0;
+    }
+
+    const minLength = Math.min(s1.length, s2.length);
+
+    for (let i = 0; i < minLength; i++) {
+      const c1 = s1[i];
+      const c2 = s2[i];
+
+      if (c1 === c2) {
+        continue;
+      }
+
+      if (c1 > c2) {
+        return 1;
+      } else if (c1 < c2) {
+        return -1;
+      }
+    }
+
+    // Okay, they've been the exact same up until now, so return the longer one
+    return s1.length > s2.length ? 1 : -1;
+  }
+
+  /**
+   * Taken from https://github.com/rpm-software-management/rpm/blob/master/rpmio/rpmvercmp.c
+   */
+  protected _compare_glob(v1: string, v2: string): number {
+    if (v1 === v2) {
+      return 0;
+    }
+
+    const matchesv1 =
+      (v1.match(alphaNumPattern) as Array<string>) ?? ([] as Array<string>);
+    const matchesv2 =
+      (v2.match(alphaNumPattern) as Array<string>) ?? ([] as Array<string>);
```

Comment:
```suggestion
    const matchesv1 = v1.match(alphaNumPattern) ?? [];
    const matchesv2 = v2.match(alphaNumPattern) ?? [];
```
ups, I don't think we need those casts